common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -16,14 +41,236 @@
 #include <netlink/genl/genl.h>
 
 #include <math.h>
-
 namespace bwl {
 
 nl80211_client_impl::nl80211_client_impl(std::unique_ptr<nl80211_socket> socket)
     : m_socket(std::move(socket))
 {
 }
 
+bool nl80211_client_impl::get_radio_info(const std::string &interface_name, sRadioInfo &radio_info)
+{
+    struct channels_ctx {
+        int last_band;
+        bool width_40;
+        bool width_80;
+        bool width_80_80;
+        bool width_160;
+    } ctx{};
+
+    radio_info = {};
+
+    if (!m_socket) {
+        LOG(ERROR) << "Socket is NULL!";
+        return false;
+    }
+
+    // Get the interface index for given interface name
+    int iface_index = if_nametoindex(interface_name.c_str());
+    if (0 == iface_index) {
+        LOG(ERROR) << "Failed to read the index of interface " << interface_name << ": "
+                   << strerror(errno);
+
+        return false;
+    }
+
+    return m_socket.get()->send_receive_msg(
+        NL80211_CMD_GET_WIPHY, 0,
+        [&](struct nl_msg *msg) -> bool {
+            nla_put_u32(msg, NL80211_ATTR_IFINDEX, iface_index);
+
+            return true;
+        },
+        [&](struct nl_msg *msg) -> bool {
+            struct nlattr *tb[NL80211_ATTR_MAX + 1];
+            struct genlmsghdr *gnlh = (struct genlmsghdr *)nlmsg_data(nlmsg_hdr(msg));
```

Comment:
No C-style cast, please use a static_cast if you still find the time/motivation to change it

Reply:
Done